MonoGame.Framework/Graphics/GraphicsDevice.cs

```diff
@@ -262,6 +278,35 @@ public DepthStencilState DepthStencilState
             }
         }
 
+        internal void ApplyState(bool applyShaders)
+        {
+            PlatformBeginApplyState();
+
+            if (_blendStateDirty)
+            {
+                // Static state properties never actually get bound;
+                // instead we use our GraphicsDevice-specific version of them.
+                var newBlendState = _blendState;
+                if (ReferenceEquals(_blendState, BlendState.Additive))
+                    newBlendState = BlendStateAdditive;
+                else if (ReferenceEquals(_blendState, BlendState.AlphaBlend))
+                    newBlendState = BlendStateAlphaBlend;
+                else if (ReferenceEquals(_blendState, BlendState.NonPremultiplied))
+                    newBlendState = BlendStateNonPremultiplied;
+                else if (ReferenceEquals(_blendState, BlendState.Opaque))
+                    newBlendState = BlendStateOpaque;

```

Comment:
I just realised that it's not done on every draw call - only when `_blendStateDirty == true`. But I still think it's worth making this change - the code becomes a bit cleaner.